packages/language-core/src/generators/template.ts

```diff
@@ -1260,7 +1261,7 @@ export function generate(
 				if (
 					(!prop.arg || (prop.arg.type === CompilerDOM.NodeTypes.SIMPLE_EXPRESSION && prop.arg.isStatic)) // isStatic
 					&& hyphenateAttr(attrNameText) === attrNameText
-					&& !nativeTags.has(node.tag)
+					&& shouldCamelize
```

Comment:
You can also now move this check to be done before the other ones as it's now the cheapest one (we can avoid `hyphenateAttr`)